.github/workflows/deploy-workflow.yml

```diff
@@ -157,33 +161,48 @@ jobs:
           --values helm_deploy/values-base.yaml \
           --namespace ${{ secrets.kube_namespace }} \
           --set datahub-frontend.fullnameOverride=${RELEASE_NAME}-${FRONTEND_FULLNAME} \
-          --set "datahub-frontend.oidcAuthentication.azureTenantId=${{ vars.TENANT_ID }}" \
-          --set "datahub-frontend.oidcAuthentication.clientId=${{ vars.CLIENT_ID }}" \
-          --set "datahub-frontend.ingress.tls[0].hosts[0]=${APP_SHORT_HOST}" \
-          --set "datahub-frontend.ingress.hosts[0].host=${APP_SHORT_HOST}" \
-          --set "datahub-frontend.ingress.annotations.external-dns\\.alpha\\.kubernetes\\.io/set-identifier=${EXT_DNS_ID}" \
+          --set datahub-frontend.oidcAuthentication.azureTenantId=${{ vars.TENANT_ID }} \
+          --set datahub-frontend.oidcAuthentication.clientId=${{ vars.CLIENT_ID }} \
+          --set datahub-frontend.ingress.tls[0].hosts[0]=${APP_SHORT_HOST} \
+          --set datahub-frontend.ingress.hosts[0].host=${APP_SHORT_HOST} \
+          --set datahub-frontend.ingress.annotations.external-dns\\.alpha\\.kubernetes\\.io/set-identifier=${EXT_DNS_ID} \
           --set acryl-datahub-actions.serviceAccount.name=${IRSA_SA} \
           --set global.elasticsearch.host=${OPENSEARCH_PROXY_HOST} \
           --set global.sql.datasource.host=${POSTGRES_HOST} \
           --set global.sql.datasource.hostForpostgresqlClient=${POSTGRES_CLIENT_HOST} \
           --set global.sql.datasource.url=${POSTGRES_URL}
 
-      - name: install datahub-monitoring helm
+      - name: allow CP prometheus scraping 
+        if: ${{ inputs.env == 'dev' }}
         shell: bash
+        id: allow-prom-scrape
         env:
           KUBE_NAMESPACE: ${{ secrets.kube_namespace }}
-          DASHBOARD_NAME: datahub-status-dashboard
         run: |
           envsubst < helm_deploy/monitoring/datahub-networkpolicy.yaml | 
             kubectl apply -f - --namespace=${KUBE_NAMESPACE}
 
-          kubectl create configmap ${DASHBOARD_NAME} \
-          --from-file=helm_deploy/monitoring/datahub-dashboard.json \
-          --dry-run \
-          --output yaml | 
-            kubectl label -f- \
+      - name: update grafana dashboard configmaps
+        if: ${{ inputs.env == 'dev' }}
+        shell: bash
+        id: update-grafana-dashboards
+        env:
+          KUBE_NAMESPACE: ${{ secrets.kube_namespace }}
+          DASHBOARDS_JSON_STRING: '{"datahub-status-dashboard": "datahub-dashboard","datahub-deployment-dashboard": "datahub-deployment-dashboard"}'
+        run: |
+          DASHBOARDS_JSON=$(jq -n -c "$DASHBOARDS_JSON_STRING")
+          echo $DASHBOARDS_JSON
+          echo $DASHBOARDS_JSON | jq -r 'to_entries | .[].key' | while read DASHBOARD; do
+            DASHBOARD_FILE=$(echo $DASHBOARDS_JSON | jq -r --arg e "${DASHBOARD}" '.[$e]')
```

Comment:
did you consider handling the different dashboards as gh actions jobs rather than looping through json?

Reply:
I considered splitting it into 2 steps, but I prefer single steps for the same purpose. I can split it into 2 steps if the jq is too opaque / distracting from the purpose.

Splitting into another job requires a re-setup of k8s credentials, and also spins up another compute instance.

Reply:
2 steps seems like a good approach to me